### PR TITLE
Change init and uninstall behavior for self hosted mode

### PR DESCRIPTION
**Pull request by amanbha**
Fixes https://github.com/dapr/cli/issues/411

### CODE REVIEW

**Comment by amanbha** [comment]
cc @orizohar @msfussell for doc changes.
@mukundansundar Could you please update decision record for this.

**Comment by mukundansundar** [comment]
I am thinking it would be better if we can put the binaries inside a `bin` directory in the default folder.

**Comment by amanbha** [comment]
@orizohar for doc changes review.

File: pkg/standalone/standalone.go [hunk 1]
```diff
@@ -87,8 +84,8 @@ type componentMetadataItem struct {
 }
 
 // Check if the previous version is already installed.
-func isBinaryInstallationRequired(binaryFilePrefix, installLocation, requestedVersion string) (bool, error) {
-	binaryPath := binaryFilePath(binaryFilePrefix, installLocation)
+func isBinaryInstallationRequired(binaryFilePrefix, installDir, requestedVersion string) (bool, error) {
```
Comment (pruthvidhodda) [inline pkg/standalone/standalone.go:87]: We aren't using requestedVersion here

File: cmd/uninstall.go [hunk 2]
```diff
@@ -53,7 +52,6 @@ var UninstallCmd = &cobra.Command{
 func init() {
 	UninstallCmd.Flags().BoolVarP(&uninstallKubernetes, "kubernetes", "k", false, "Uninstall Dapr from a Kubernetes cluster")
 	UninstallCmd.Flags().BoolVar(&uninstallAll, "all", false, "Remove Redis container in addition to actor placement container")
```
Comment (mukundansundar) [inline cmd/uninstall.go:54]: Can you add the .dapr folder, redis and zipkin containers?

File: README.md [hunk 3]
```diff
@@ -62,7 +62,7 @@ In self-hosted mode, dapr can be initialized using the CLI  with the placement,
 #### Initialize Dapr
 ([Prerequisite](#Prerequisites): Docker is available in the environment - recommended)
 
-Use the init command to initialize Dapr. On init, multiple default configuration files and containers are installed along with the dapr runtime binary. Dapr runtime binary is installed under $HOME/.dapr for Mac, Linux and %USERPROFILE%\.dapr for Windows.
+Use the init command to initialize Dapr. On init, multiple default configuration files and containers are installed along with the dapr runtime binary. Dapr runtime binary is installed under $HOME/.dapr/bin for Mac, Linux and %USERPROFILE%\.dapr/bin for Windows.
```
Comment (mukundansundar) [inline README.md:65]: nitpick: `\`

**Comment by mukundansundar** [comment]
Also #405

**Pull request merged**